bio/bwa/mem/wrapper.py

```diff
@@ -36,32 +48,30 @@
 
 elif sort == "samtools":
 
-    # Sort alignments using samtools sort.
-    pipe_cmd = "samtools sort {sort_extra} -o {snakemake.output[0]} -"
-
     # Add name flag if needed.
     if sort_order == "queryname":
         sort_extra += " -n"
 
-    prefix = path.splitext(snakemake.output[0])[0]
-    sort_extra += " -T " + prefix + ".tmp"
+    # Sort alignments using samtools sort.
+    pipe_cmd = "samtools sort -T {tmp}/{tmp_prefix}.tmp {sort_extra} -o {snakemake.output[0]} -"
```

Comment:
In line with my above comment, I would simply go for:
```suggestion
    pipe_cmd = "samtools sort -T {tmp} {sort_extra} -o {snakemake.output[0]} -"
```